lib/mix/lib/mix/tasks/compile.app.ex

```diff
@@ -44,14 +44,42 @@ defmodule Mix.Tasks.Compile.App do
       The application environment is one of the most common ways
       to configure applications.
 
-  Let's see an example `application/0` function:
+    * `:start_phases` - specifies a list of phases and their arguments
+      to be called after the application is started and before any
+      included applications are started.
+
+    * `:included_applications` - specifies a list of applications
+      that will be included in the application. It is the responsability of
+      the primary application to start the supervision tree of any included
+      applications, as only the primary application will be started. A process
+      in an included application considers itself belonging to the
+      primary application.
+
+  Let's see an example `MyApp.application/0` function:
 
       def application do
         [mod: {MyApp, []},
+         start_phases: [init: [], go: []],
          env: [default: :value],
-         applications: [:crypto]]
+         applications: [:crypto],
+         included_applications: [MyIncludedApp]]
+      end
+
+  And an example `MyIncludedApp.application/0` function:
+
+      def application do
+        [mod: {MyIncludedApp, []},
+         start_phases: [go: []]]
       end
 
+  In this example, the order that the application callbacks are called in is:
```

Comment:
I'm not sure if this is clear enough on what is going on with the ordering of the callbacks.  It's based on the example at the bottom of the page here (http://erlang.org/doc/design_principles/included_applications.html#id83214)